website/sales/views.py

```diff
@@ -0,0 +1,41 @@
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import get_object_or_404, redirect, render
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext_lazy as _
+from django.views import View
+
+from sales.models.order import Order
+
+
+@method_decorator(login_required, name="dispatch")
+class OrderPaymentView(View):
```

Comment:
Looks like you really need a error/success page for the redirect. Loads of TODOs here :)

Reply:
Yes. Also see #1616

Reply:
I saw the issue already, but should we have that block this PR?

Reply:
No not necessarily. 

I'm also wondering what you guys think about the exact payment flow. Currently there is a 2-step flow because I'm using the payment templatetag (see the comments in the code)

Reply:
I tried this and I would say that it is somewhat cumbersome, seems to me that you could just redirect to the actual payment page instead.

Reply:
The sad thing is that the normal payment confirmation page is a form that requires a `POST`. But a QR code can't do that. So I will need to create a special payment confirmation page for this :(

Reply:
You could add some Javascript to the rendered page that automatically submits the form if JS is enabled to skip the step? Maybe that is an option.

I was also thinking about adding this page to thadmin, but that would make it harder because then you would need to login to Thadmin also instead of just the website. Regular users should probably not be bothered with another website.

Reply:
I also thought about the javascript option yeah, I think I will be doing that.

I think `thadmin` shouldn't be for regular users. Thadmin is just a nice frontend to optimize the UI/UX for shift managers. The whole system should operate without `thadmin` as well (which it does right now, you can just use the Django admin for everything and it will work fine)

Reply:
I think we have reached a good consensus 👍